backend.native/tests/runtime/text/utf8.kt

```diff
@@ -70,51 +139,115 @@ fun <T: Any> checkThrows(e: KClass<T>, string: String, action: () -> Unit) {
+    checkOutOfBoundsUtf16to8Throwing(e, string, start, size)
+}
+
+
+// utils for checking invalid-bounds-exception thrown by UTF-8 -> UTF-16 conversion
+fun <T: Any> checkOutOfBoundsZeroTerminatedUtf8to16Replacing(e: KClass<T>, string: String, byteArray: ByteArray, start: Int, size: Int) {
+    checkThrows(e, string) { byteArray.stringFromUtf8(start, size) }
+    checkThrows(e, string) { byteArray.decodeZeroTerminatedToString(start, start + size) }
+}
+fun <T: Any> checkOutOfBoundsUtf8to16Replacing(e: KClass<T>, string: String, byteArray: ByteArray, start: Int, size: Int) {
+    checkThrows(e, string) { byteArray.decodeToString(start, start + size) }
+    checkOutOfBoundsZeroTerminatedUtf8to16Replacing(e, string, byteArray, start, size)
+}
+fun <T: Any> checkOutOfBoundsZeroTerminatedUtf8to16Throwing(e: KClass<T>, string: String, byteArray: ByteArray, start: Int, size: Int) {
+    checkThrows(e, string) { byteArray.stringFromUtf8OrThrow(start, size) }
+    checkThrows(e, string) { byteArray.decodeZeroTerminatedToString(start, start + size, true) }
+}
+fun <T: Any> checkOutOfBoundsUtf8to16Throwing(e: KClass<T>, string: String, byteArray: ByteArray, start: Int, size: Int) {
+    checkThrows(e, string) { byteArray.decodeToString(start, start + size, true) }
+    checkOutOfBoundsZeroTerminatedUtf8to16Throwing(e, string, byteArray, start, size)
+}
+fun <T: Any> checkOutOfBoundsUtf8to16(e: KClass<T>, string: String, byteArray: ByteArray, start: Int, size: Int) {
+    checkOutOfBoundsUtf8to16Replacing(e, string, byteArray, start, size)
+    checkOutOfBoundsUtf8to16Throwing(e, string, byteArray, start, size)
+}
+fun <T: Any> checkOutOfBoundsZeroTerminatedUtf8to16(e: KClass<T>, string: String, byteArray: ByteArray, start: Int, size: Int) {
+    checkOutOfBoundsZeroTerminatedUtf8to16Replacing(e, string, byteArray, start, size)
+    checkOutOfBoundsZeroTerminatedUtf8to16Throwing(e, string, byteArray, start, size)
+}
+
+
+// util for performing action on result of UTF-8 -> UTF-16 conversion
+fun convertUtf8to16(byteArray: ByteArray, action: (String) -> Unit) {
+    byteArray.stringFromUtf8().let { action(it) }
+    byteArray.decodeToString().let { action(it) }
+    byteArray.decodeZeroTerminatedToString().let { action(it) }
+}
 // endregion
 
 // region UTF-16 -> UTF-8 conversion
 fun test16to8() {
-    // Test manual conversion with replacement.
     // Valid strings.
-    checkUtf16to8Replacing("Hello", intArrayOf('H'.toInt(), 'e'.toInt(), 'l'.toInt(), 'l'.toInt(), 'o'.toInt()))
-    checkUtf16to8Replacing("Привет", intArrayOf(-48, -97, -47, -128, -48, -72, -48, -78, -48, -75, -47, -126))
-    checkUtf16to8Replacing("\uD800\uDC00", intArrayOf(-16, -112, -128, -128))
-    checkUtf16to8Replacing("", intArrayOf())
+    checkValidUtf16to8("Hello", intArrayOf('H'.toInt(), 'e'.toInt(), 'l'.toInt(), 'l'.toInt(), 'o'.toInt()))
+    checkValidUtf16to8("Привет", intArrayOf(-48, -97, -47, -128, -48, -72, -48, -78, -48, -75, -47, -126))
```

Comment:
Why not proper uintArrayOf() with hex unsigned values?

Reply:
or even `ubyteArrayOf`

Reply:
Tests here were not modified. I just refactored the utils to run a test for multiple methods, 
e.g. `toUtf8()`, `encodeToByteArray`, `toUtf8OrThrow` and `encodeToByteArray(throwOnInvalidSequence = true)` must all return the same UFT-8 bytes for a valid string.

Converting all these values to hex form seems not to be time-worthy.